docs: restructure and expand SLIM documentation (#1789)
agntcy/slim: docs/content/slim/components/data-plane/index.md

## Responsibilities

- **Message routing** — delivers messages to named endpoints using anycast (any instance of a service) or unicast (a specific instance) addressing
- **Connection management** — accepts inbound connections from applications over gRPC or WebSocket; maintains outbound connections to peer SLIM nodes to form a mesh
- **Transport security** — secures connections with TLS or mTLS between applications and nodes, and between peer nodes
- **Controller integration** — registers with the SLIM Controller and receives route and connection configuration updates via the Controller's southbound API
- **Observability** — exports traces and metrics via OpenTelemetry

## Architecture

Internally the Data Plane is composed of two main layers:

- **Service layer** — bootstraps the process, owns the transports (gRPC and WebSocket listeners), and spawns per-connection tasks. Each accepted connection runs in its own async task and feeds received messages into the core routing engine.

> - **Datapath layer** — the `MessageProcessor` is the central hub. It owns a `ConnectionTable` (all active connections) and a `SubscriptionTable` (name-to-connection mappings that drive routing). A `Forwarder` executes the fanout: given a message, it looks up matching subscriptions and writes the message to each matched connection.

Reviewer: Technically the message processor owns the connections, the listeners and the tasks as well. The service layer owns the message processor itself and exposes some APIs to create connections and servers in the message processor.

### Message Flow

1. A connection task receives a message and performs basic validation: HMAC integrity check and TTL decrement. Messages that fail validation or whose TTL reaches zero are dropped.
2. The message type determines dispatch: **subscribe** updates the subscription table, **publish** triggers a routing table lookup, **link** (peer control) updates the connection table.

> 3. For publish messages the Forwarder looks up all subscriptions matching the destination name, then round-robins across the matching connections and writes the message to each.

Reviewer: ```suggestion
3. For publish messages, the Forwarder looks up all subscriptions matching the destination name and then either selects a single connection using a round-robin approach, or sends the message to all matching connections.
```

### Connection Types

The Data Plane distinguishes four connection categories:

| Type | Description |
|------|-------------|
| **Local** | In-process applications connected via the SDK |

> | **Remote** | Applications connected over the network (gRPC or WebSocket) |

Reviewer: ```suggestion
| **Edge** | Applications connected over the network (gRPC or WebSocket) |
```

Reviewer 2: This is the first hop connection from the application to the network.

| **Peer** | Other Data Plane nodes in the same deployment (replica peers) |

> | **Edge** | Data Plane nodes in remote deployments (inter-cluster links) |

Reviewer: ```suggestion
| **Remote** | Data Plane nodes in remote deployments (inter-cluster links) |
```
